db/migrate/20220219083831_add_users_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUsersRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: foreign_key: {to_table: :users}
```

Comment:
[OPTIONAL]
- I suggest you make the reference as `:author` instead of `:user`, so you don't conflict the `user_id` for users